lib/section_slug_synchroniser.rb

```diff
@@ -0,0 +1,75 @@
+class SectionSlugSynchroniser
+  attr_reader :logger, :manual, :sections
+
+  def initialize(manual_slug, logger = STDOUT)
+    @logger = logger
+    manual_record = ManualRecord.where(slug: manual_slug).last
+    @manual = Manual.find(manual_record.manual_id, user)
+    @sections = manual.sections
+  end
+
+  def report
+    amendments, conflicts = analyse_sections
+
+    if amendments.any?
+      log "The following sections can be reslugged:"
+      amendments.each { |k, v| log "'#{section_slug(k)}' to '#{section_slug(v)}'" }
+    end
+
+    if conflicts.any?
+      log "The following sections cannot be reslugged:"
+      conflicts.each do |k, v|
+        log "'#{section_slug(k)}' would change to '#{section_slug(v)}' but this is already in use."
+      end
+    end
+  end
+
+  def synchronise
+    amendments, = analyse_sections
```

Comment:
might be an improvement to have this as `analyse_sections.first` or similar, the trailing comma looks a bit like a typo